src/collective/communicator.cc

```diff
@@ -0,0 +1,59 @@
+/*!
+ * Copyright 2022 XGBoost contributors
+ */
+#include "communicator.h"
+
+#include "rabit_communicator.h"
+
+#if defined(XGBOOST_USE_FEDERATED)
+#include "../../plugin/federated/federated_communicator.h"
+#endif
+
+namespace xgboost {
+namespace collective {
+
+thread_local std::unique_ptr<Communicator> Communicator::communicator_{};
+thread_local CommunicatorType Communicator::type_{};
+
+void Communicator::Init(Json const& config) {
+  if (communicator_) {
+    LOG(FATAL) << "Communicator can only be initialized once.";
+  }
+
+  auto type = GetTypeFromEnv();
+  auto const arg = GetTypeFromConfig(config);
+  if (arg != CommunicatorType::kUnknown) {
+    type = arg;
+  }
+  if (type == CommunicatorType::kUnknown) {
+    // Default to Rabit if unspecified.
+    type = CommunicatorType::kRabit;
+  }
+  type_ = type;
+  switch (type) {
+    case CommunicatorType::kRabit: {
+      communicator_.reset(RabitCommunicator::Create(config));
+      break;
+    }
+    case CommunicatorType::kMPI:
+      LOG(FATAL) << "Not implemented yet.";
```

Comment:
If you don't plan to implement it, we can just remove it.

Reply:
Do you know if people actually use the current MPI integration with Rabit? It's probably not too hard to add it if there is demand for it.

Reply:
Removed for now.